### PR TITLE
Prepare v0.5 release

### PR DESCRIPTION
**Pull request by irakliyk**
This PR prepares the crates for v0.5 release. Specifically:

- Updates versions of all crates to v0.5.
- Updates versions of outdated dependences (there were a couple).
- Updates minimum required Rust version to v1.67.
- Updates the CHANGELOG.

### CODE REVIEW

**Comment by irakliyk** [comment]
@Nashtare, @Al-Kindi-0 - would appreciate a review. Thank you!

File: air/Cargo.toml [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 readme = "README.md"
 license = "MIT"
 repository = "https://github.com/novifinancial/winterfell"
-documentation = "https://docs.rs/winter-air/0.4.2"
+documentation = "https://docs.rs/winter-air/0.5.0"
 categories = ["cryptography", "no-std"]
 keywords = ["crypto", "arithmetization", "air"]
 edition = "2021"
-rust-version = "1.60"
+rust-version = "1.67"
```
Comment (Nashtare) [inline air/Cargo.toml:13]: Curious about the toolchain change directly up to 1.67. I assumed there was a breaking change but I'm on 1.66 locally and everything runs smoothly.

**Review by Nashtare — COMMENTED** [review]
LGTM, modulo waiting for #146 and #150 as mentioned in the CHANGELOG.
Also have a tiny question about the need for raising MSRV to 1.67.

File: air/Cargo.toml (hunk 1 above)
Comment (irakliyk) [reply air/Cargo.toml:13]: There was no real need to push it that far (and I think 1.60 would have also worked fine). I do notice that there are minor performance improvements with each new version of Rust - so, didn't want it to fall too far behind. I can change it to 1.66 if that works better though.

Comment (Nashtare) [reply air/Cargo.toml:13]: No need to downgrade it to 1.66 🙂 I was just curious about the rationale behind the bump.

**Pull request merged**